apps/src/music/blockly/MusicBlocklyWorkspace.ts

```diff
@@ -569,9 +569,33 @@ export default class MusicBlocklyWorkspace {
     if (this.blockMode) {
       const existingToolbox = getToolbox(this.blockMode, this.toolbox);
       existingToolbox.contents = existingToolbox.contents.concat(blockList);
-      (this.workspace as GoogleBlockly.WorkspaceSvg).updateToolbox(
-        existingToolbox
-      );
+      const workspace = this.workspace as GoogleBlockly.WorkspaceSvg;
+      workspace.updateToolbox(existingToolbox);
+
+      if (workspace.RTL) {
+        // When the flyout is dynamically populated, the flyout width can increase,
+        // thereby overlapping start blocks in RTL. If this happens, we move the
+        // blocks back to the left
+        // Relates to https://github.com/google/blockly/issues/8637
+        const flyout = workspace.getFlyout();
+        const metricsManager = workspace.getMetricsManager();
+        const flyoutWidth = flyout?.getWidth() || 0;
+
+        if (flyoutWidth) {
+          const {left: contentLeft, width: contentWidth} =
+            metricsManager.getContentMetrics();
+          const viewWidth = metricsManager.getViewMetrics().width;
+
+          const contentRight = contentLeft + contentWidth;
+          const overlapAmount = contentRight - viewWidth + 20; // 20 is the expected margin.
+
+          if (contentRight >= viewWidth) {
```

Comment:
Is the expected margin how much space we expect between the flyout and the blocks?

Should we incorporate the margin into deciding whether to do this adjustment?  i.e. it seems that right now, we don't do the adjustment when the content is less than 20 pixels from the flyout?